wordle-web/pages/wordEditor.vue

```diff
@@ -0,0 +1,364 @@
+  </v-card>
+  <v-dialog v-model="addWordDialog" width="400">
+    <v-card height="200">
+      <!--<v-text-field label="Word" v-model="wordToAdd" />-->
+      <v-autocomplete
+        v-model="wordToAdd"
+        :items="words!.map(w => w.word)"
+        label="Add Word"></v-autocomplete>
+      <v-btn
+        @click="
+          addWord();
+          addWordDialog = false;
+        "
+        >Submit</v-btn
+      >
+    </v-card>
+  </v-dialog>
+</template>
+
+<script setup lang="ts">
+import Axios from 'axios';
+import { TokenService, key } from '~/scripts/tokenService';
+
+interface Word {
+  word: string;
+  isCommonWord: boolean;
+}
+
+enum CommonWordSortOrder {
+  Ascending = 0,
+  Descending = 1,
+  Unspecified = 2,
+}
+
+enum CommonRadio {
+  Both = 0,
+  Common = 1,
+  Uncommon = 2,
+}
+
+const isLoading = ref<boolean>(true);
+const words = ref<Array<Word>>();
+const wordsCopy = ref<Array<Word>>();
+const isCommonWordChanged = ref<boolean[]>([]);
+const tokenService = ref(inject(key));
+const isWordOrderAscending = ref(true);
+const commonWordOrder = ref(CommonWordSortOrder.Unspecified);
+const addWordDialog = ref(false);
+const wordToAdd = ref('');
+const commonRadio = ref(CommonRadio.Both);
+
+const itemsPerPage = ref(10);
+const curretnPage = ref(1);
+
+function paginatedWords() {
+  const start = (curretnPage.value - 1) * itemsPerPage.value;
+  const end = start + itemsPerPage.value;
+  return words.value?.slice(start, end);
+}
+function pageCount() {
+  return Math.ceil(words.value!.length / itemsPerPage.value);
+}
+
+const wordPerPage = ref(10);
+const wordText = ref('');
+const pageNum = ref(1);
+const countPages = ref(0);
+
+type MoreWordsDto = {
+  words: Word[];
+  countPages: number;
+};
+
+//GetMoreWords(int wordCount, string? word, int pages)
+const pageDisplay = (pageWord: number, text: string, page: number) => {
+  wordPerPage.value = pageWord;
+  wordText.value = text;
+  pageNum.value = page;
+  Axios.get(
+    `word/GetMoreWords?wordCount=${wordPerPage.value}&word=${wordText.value}&pages=${pageNum.value}`
+  ).then(result => {
+    const data = result.data as MoreWordsDto;
+    words.value = result.data;
+    countPages.value = data.countPages;
+    console.log(result);
+  });
+};
+
+const isMotU = computed(() => tokenService.value?.getMotU());
+const isOlderThanTwentyOne = computed(() =>
+  tokenService.value?.isOlderThanTwentyOne()
+);
+
+watch([commonRadio], () => {
+  sortWords(wordsCopy.value!);
+  switch (commonRadio.value) {
+    case CommonRadio.Both:
+      words.value = wordsCopy.value;
+      break;
+    case CommonRadio.Common:
+      words.value = wordsCopy.value?.filter(word => word.isCommonWord);
+      break;
+    case CommonRadio.Uncommon:
+      words.value = wordsCopy.value?.filter(word => !word.isCommonWord);
+      break;
+  }
+});
+
+try {
+  const gameUrl = 'word/getallwords';
+  Axios.get(gameUrl).then(response => {
+    words.value = response.data;
+    wordsCopy.value = response.data;

```

Comment:
Why are you copying the words twice? It seems to be based on filtering the words. This could work but copying the words each time feels excessive. You seem to be only using it in your filter method.

Reply:
After thinking about it, you're very right. My original thought was to ensure no data is lost. The main 'words' ref is to display the data (filtered or unfiltered), and the copy is to maintain the full list. My idea now is to save each list once (Both common/uncommon, Only common, and Only uncommon). What do you think? I'm open to suggestions.